src/coreclr/src/gc/gcdesc.h

```diff
@@ -199,28 +199,38 @@ class CGCDesc
     static size_t GetNumPointers (MethodTable* pMT, size_t ObjectSize, size_t NumComponents)
     {
         size_t NumOfPointers = 0;
-        CGCDesc* map = GetCGCDescFromMT(pMT);
-        CGCDescSeries* cur = map->GetHighestSeries();
-        ptrdiff_t cnt = (ptrdiff_t) map->GetNumSeries();
 
-        if (cnt > 0)
+        if (pMT->ContainsPointers())
         {
-            CGCDescSeries* last = map->GetLowestSeries();
-            while (cur >= last)
+            CGCDesc* map = GetCGCDescFromMT(pMT);
+            CGCDescSeries* cur = map->GetHighestSeries();
+            ptrdiff_t cnt = (ptrdiff_t)map->GetNumSeries();
+
+            if (cnt >= 0)
             {
-                NumOfPointers += (cur->GetSeriesSize() + ObjectSize) / sizeof(JSlot);
-                cur--;
+                CGCDescSeries* last = map->GetLowestSeries();
+                do
+                {
+                    NumOfPointers += (cur->GetSeriesSize() + ObjectSize) / sizeof(JSlot);
```

Comment:
Will `cur` never be `null`? The previous code would have handled that without issue if `last` used some sentinel value, but now we dereference `cur` all the time.

Reply:
It cannot be ever null.